canonical/vanilla-framework
# Fix styling issues of meganav on small screens

## Done

Addresses styling and scrolling issues in mobile version of mega navigation

- fixes scroll when there are many items in the nav on mobile
- removes visual glitches when scrolling
- makes styling inside mobile nav more consistent

Fixes: https://warthogs.atlassian.net/browse/WD-13787

## QA

- Example (test in mobile view to see sliding nav instead of full width dropdowns):
https://vanilla-framework-5110.demos.haus/docs/examples/patterns/navigation/dropdown-full-width?theme=light
- Note: scope of the PR is focused on the functionality of the sliding navigation strucutre (scrolling, sliding animation, etc) - styling of individual items inside of mobile navigation (such as quick links, download buttons, etc) is not part of this PR (they are added only for demo purposes)
- Test:
  - Click around the nested levels of the navigation, make sure it animates as expected
  - Check if scroll works as expected
  - Verify that desktop view didn't get broken
- Check other related examples:
  - Reduced navigation: should have the same content and work the same:  https://vanilla-framework-5110.demos.haus/docs/examples/patterns/navigation/reduced?theme=light
  - Sliding navigation (should still work as well): https://vanilla-framework-5110.demos.haus/docs/examples/patterns/navigation/sliding-dark?theme=light

### QA Note

In Vanilla examples view the bottom toolbar is covering some space on the bottom of the screen. In normal view it would not be there and there will be more of the dropdown visible.

<img width="563" alt="image" src="https://github.com/user-attachments/assets/de1bc7da-ba7c-45c3-a556-c8b0172ffa59">


### Implementation note

This PR introduces 2 new class names required by JS.

`js-navigation-sliding-panel` needs to be added on the elements with `p-navigation__dropdown` CSS class name (so that they are correctly targeted when selecting a dropdown to activate), it also needs to be added on a top level `p-navigation__items` element (the one containing first level of mobile sliding navigation items)

`js-navigation-dropdown-toggle` needs to be added on elements with `p-navigation__item--dropdown-toggle` class name.
